frontend/src/lib/components/Editor.svelte

```diff
@@ -65,6 +125,54 @@
 	<div bind:this={previewWindow} class="preview-pane"></div>
 </div>
 
+{#if showCommitModal}
+	<div
+		on:click={() => {
+			showCommitModal = false;
+		}}
+		on:keydown={(e) => {
+			if (e.key === 'Escape') {
+				showCommitModal = false;
+			}
+		}}
+		role="button"
+		tabindex="0"
+		class="commit-modal-backdrop"
+	></div>
+	<div id="commitModal" class="commit-modal" bind:this={commitModal}>
+		<div class="commit-modal-content">
+			<svg
+				on:click={() => {
+					showCommitModal = false;
+				}}
+				on:keypress={() => {
+					showCommitModal = false;
+				}}
+				class="commit-modal-close"
+				role="button"
+				tabindex="0"
+				xmlns="http://www.w3.org/2000/svg"
+				height="24px"
+				viewBox="0 -960 960 960"
+				width="24px"
+				fill="#e8eaed"
+			>
+				<path
+					d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224-224 224Z"
+				/>
+			</svg>
+			<h2>Confirm changes before committing:</h2>
+			<h5>Enter a commit message (optional)</h5>
+			<input type="text" id="commitMessage" placeholder="Enter your commit message here" bind:this={commitMessageInput} maxlength="500" on:input={updateCharCount}>
```

Comment:
I don't believe the id here is needed, and if it is, it should be kebab-case, not camelCase